app/src/main/java/com/github/multimatum_team/multimatum/service/ProcrastinationDetectorService.kt

```diff
@@ -17,38 +22,131 @@ import kotlin.math.abs
 /**
  * This service detects the movements of the phone to deduce whether the user is
  * working or not, and can display a reminder if necessary
+ *
+ * Adapted from https://robertohuertas.com/2019/06/29/android_foreground_services/
  */
 @AndroidEntryPoint
 class ProcrastinationDetectorService : Service(), SensorEventListener {
     @Inject
     lateinit var sensorManager: SensorManager
 
-    // TODO make it remain enabled when app is stopped
+    private var isServiceStarted = false
+    private var wakeLock: PowerManager.WakeLock? = null
 
     // data relative to the last time a movement was detected
     private var lastDetectionTimestamp: Long = 0L
     private var lastPosition: Array<Float>? = null  // null only at initialization
 
     private val binder = PdsBinder()
 
+    override fun onBind(intent: Intent): IBinder {
+        return binder
+    }
+
+    // also used to stop the service, when action is STOP_ACTION
```

Comment:
I agree that this comment is unclear and I will change it. However, I tried to avoid the need for comments by making the code modular. As you can see, functions are short and their names describe what they do, instead of comments. So I do not really understand what additional comments I should write...